packages/rancher-monitoring/rancher-monitoring/generated-changes/patch/templates/prometheus/rules-1.14/etcd.yaml.patch

```diff
@@ -10,6 +10,24 @@
  apiVersion: monitoring.coreos.com/v1
  kind: PrometheusRule
  metadata:
+@@ -58,7 +59,7 @@
+       annotations:
+         message: 'etcd cluster "{{`{{`}} $labels.job {{`}}`}}": {{`{{`}} $value {{`}}`}}% of requests for {{`{{`}} $labels.grpc_method {{`}}`}} failed on etcd instance {{`{{`}} $labels.instance {{`}}`}}.'
+       expr: |-
+-        100 * sum(rate(grpc_server_handled_total{job=~".*etcd.*", grpc_code!="OK"}[5m])) BY (job, instance, grpc_service, grpc_method)
++        100 * sum(rate(grpc_server_handled_total{job=~".*etcd.*", grpc_code=~"Unknown|FailedPrecondition|ResourceExhausted|Internal|Unavailable|DataLoss|DeadlineExceeded"}[5m])) BY (job, instance, grpc_service, grpc_method)
+           /
+         sum(rate(grpc_server_handled_total{job=~".*etcd.*"}[5m])) BY (job, instance, grpc_service, grpc_method)
+           > 1
+@@ -72,7 +73,7 @@
+       annotations:
+         message: 'etcd cluster "{{`{{`}} $labels.job {{`}}`}}": {{`{{`}} $value {{`}}`}}% of requests for {{`{{`}} $labels.grpc_method {{`}}`}} failed on etcd instance {{`{{`}} $labels.instance {{`}}`}}.'
+       expr: |-
+-        100 * sum(rate(grpc_server_handled_total{job=~".*etcd.*", grpc_code!="OK"}[5m])) BY (job, instance, grpc_service, grpc_method)
++        100 * sum(rate(grpc_server_handled_total{job=~".*etcd.*", grpc_code=~"Unknown|FailedPrecondition|ResourceExhausted|Internal|Unavailable|DataLoss|DeadlineExceeded"}[5m])) BY (job, instance, grpc_service, grpc_method)
```

Comment:
are these the only two rules that need to have modified queries or did you go through all the changes in https://github.com/etcd-io/etcd/blob/main/contrib/mixin/mixin.libsonnet to ensure that our queries match what is the latest in upstream?

If we are patching this fix, we might as well patch the fix for this for all the libsonnet stuff to get the latest from etcd.

Also, from a code perspective, we should leave a comment here that this was manually patched by Rancher outside of the scripts in order to reflect the latest that etcd has to offer.

Reply:
I went through the other changes in the file, none of them really had any major modifications to any of the queries. And I also went back and added a comment just so it's clear that we made the change ourselves